### PR TITLE
Added naming of backup engine threads

### PR DESCRIPTION
**Pull request by s-sharan**
Changed the naming of backup engine threads from "ldb" to "backup_engine"
Test Plan:
screen shot 2017-03-29 at 10 23 46 am

### CODE REVIEW

**Comment by ajkr** [comment]
In this case the background worker thread is actually thread 19. Thread 1 is the user thread who's calling CreateNewBackup, we don't want to rename this one.

**Comment by ajkr** [comment]
These are the threads we want to rename: https://github.com/facebook/rocksdb/blob/master/utilities/backupable/backupable_db.cc#L649-L660

**Comment by s-sharan** [comment]
Fixed it

backup_engine

File: utilities/backupable/backupable_db.cc [hunk 1]
```diff
@@ -647,6 +647,7 @@ Status BackupEngineImpl::Initialize() {
   // background
   for (int t = 0; t < options_.max_background_operations; t++) {
     threads_.emplace_back([this]() {
+      pthread_setname_np(pthread_self(), "backup_engine");
```
Comment (ajkr) [inline utilities/backupable/backupable_db.cc]: I think it's glibc-specific, maybe conditionally compile this line using `#ifdef OS_LINUX`

**Pull request closed without merging**